vesc_hw_interface/launch/test.xacro

```diff
@@ -7,7 +7,8 @@
     <child link="vesc_link"/>
     <origin xyz="0 0 0" rpy="0 0 0"/>
     <axis xyz="0 0 1"/>
-    <limit effort="100" lower="-100" upper="100" velocity="100"/>
+    <limit effort="100" lower="-100" upper="100" velocity="1000"/>
+    <safety_controller k_velocity="10" k_position="15" soft_lower_limit="-2.0" soft_upper_limit="0.5" />
```

Comment:
Is this a test parameter that you would like to test in the future, or is this parameter tested already?
If the parameter is tested already, do you have any recommended PID gains to test it?

Reply:
I have tested that it does not rotate faster than the speed set in ```k_velocity``` of safety_controller.(with wheel controller)
I don't know what the other parameters are.
I tested movero hub motor with the params below.

```yaml
    <rosparam>
      joint_name: vesc_joint
      command_mode: velocity_duty
      joint_type: continuous
      port: /dev/ttyACM0
      num_rotor_poles: 20
      gear_ratio: 0.230769
      torque_const: 1
      motor:
        Kp: 0.02
        Ki: 0.01
        Kd: 0.03
        i_clamp: 0.2
        duty_limiter: 0.3
        antiwindup: true
        control_rate: 100
    </rosparam>
```

Reply:
Sorry for the lack of clarification; I meant the PID gain passed given to ros_controller.

I believe safety_controller tags are used if we have something set up like vesc hardware interface to JointVelocityInterface but control with PositionController.
ex:
```xml
  <arg name="model" default="$(find vesc_hw_interface)/launch/test.xacro"/>
  <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>
  <!-- Boot hardware interfaces -->
  <node name="vesc_hw_interface_node" pkg="vesc_hw_interface" type="vesc_hw_interface_node" output="screen">
    <rosparam>
      joint_name: vesc_joint
      command_mode: velocity
      port: /dev/ttyUSB0
      num_rotor_poles: 20
      gear_ratio: 1
      torque_const: 1
    </rosparam>
  </node>

  <!-- Boot ros_controllers -->
  <rosparam>
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 100

    joint_position_controller:
      type: position_controllers/JointVelocityController
      publish_rate: 100
      base_frame_id: base_link
      joint: vesc_joint
     pid: {p: 100.0, i: 0.01, d: 10.0}
  </rosparam>
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_state_controller joint_position_controller"/>

```

Reply:
Note that VescServoController and VescWheelController is setup as JointPositionController thus, safety_controller tag is not used

Reply:
I'm leaving the changes here since this is just a sample parameter. Will reopen/tune if necessary.